### ISSUE TITLE
About Straight Through Gumbel Softmax

### ISSUE THREAD

**Issue by gicheonkang** [yuleiniu/rva#10]
Hi, thank you for sharing you code.
In your paper, You wrote that rva model obtain one-hot vector using Eq. 19. Strictly speaking, I think it is not correct. 

Rva model obtain one-hot vector one_hot(argmax(Eq. 20)) not Eq. 19.
In other words, softmax operation is always used regardless of the argument (soft / hard). 

For more details, please refer to the F.gumbel_softmax code as below:
https://pytorch.org/docs/stable/_modules/torch/nn/functional.html

**Comment by yuleiniu** [comment]
Thanks for your interest. The results of argmax operation is not affected by softmax or the constant \tau.
